tests/core/test_state_sync.py

```diff
@@ -1250,29 +1250,65 @@ def test_get_version(state_sync: EngineAdapterStateSync) -> None:
         state_sync.get_versions()
     state_sync.get_versions(validate=False)
 
+
+def test_version_sqlmesh(state_sync: EngineAdapterStateSync) -> None:
+    from sqlmesh import __version__ as SQLMESH_VERSION
+    from sqlmesh import __version_tuple__ as SQLMESH_VERSION_TUPLE
+
+    # patch version sqlmesh doesn't matter
+    major, minor, patch, *_ = SQLMESH_VERSION_TUPLE
+    sqlmesh_version_patch_bump = f"{major}.{minor}.{int(patch) + 1}"
+    state_sync._update_versions(sqlmesh_version=sqlmesh_version_patch_bump)
+    state_sync.get_versions(validate=False)
+
+    # sqlmesh version is behind
+    sqlmesh_version_minor_bump = f"{major}.{int(minor) + 1}.{patch}"
+    error = (
+        rf"SQLMesh \(local\) is using version '{SQLMESH_VERSION}' which is behind '{sqlmesh_version_minor_bump}' \(remote\). "
+        rf"""Please upgrade SQLMesh \('pip install --upgrade "sqlmesh=={sqlmesh_version_minor_bump}"' command\)."""
+    )
+    state_sync._update_versions(sqlmesh_version=sqlmesh_version_minor_bump)
+    with pytest.raises(SQLMeshError, match=error):
+        state_sync.get_versions()
+    state_sync.get_versions(validate=False)
+
+    # sqlmesh version is ahead
+    sqlmesh_version_minor_decrease = f"{major}.{int(minor) - 1}.{patch}"
+    error = rf"SQLMesh \(local\) is using version '{SQLMESH_VERSION}' which is ahead of '{sqlmesh_version_minor_decrease}'"
+    state_sync._update_versions(sqlmesh_version=sqlmesh_version_minor_decrease)
+    with pytest.raises(SQLMeshError, match=error):
+        state_sync.get_versions()
+    state_sync.get_versions(validate=False)
+
+
+def test_version_sqlglot(state_sync: EngineAdapterStateSync) -> None:
     # patch version sqlglot doesn't matter
     major, minor, patch, *_ = SQLGLOT_VERSION.split(".")
     sqlglot_version = f"{major}.{minor}.{int(patch) + 1}"
     state_sync._update_versions(sqlglot_version=sqlglot_version)
     state_sync.get_versions(validate=False)
 
-    # sqlglot version is behind, always raise
+    # sqlglot version is behind
     sqlglot_version = f"{major}.{int(minor) + 1}.{patch}"
     error = (
         rf"SQLGlot \(local\) is using version '{SQLGLOT_VERSION}' which is behind '{sqlglot_version}' \(remote\). "
         rf"""Please upgrade SQLGlot \('pip install --upgrade "sqlglot=={sqlglot_version}"' command\)."""
     )
     state_sync._update_versions(sqlglot_version=sqlglot_version)
+    with pytest.raises(SQLMeshError, match=error):
+        state_sync.get_versions()
```

Comment:
This looks like a typo in this test. `error` was defined but never used. Updated to check for it similar to what is done below.